airflow/models/dagrun.py

```diff
@@ -91,6 +103,16 @@ class DagRun(Base, LoggingMixin):
         UniqueConstraint('dag_id', 'execution_date', name='dag_run_dag_id_execution_date_key'),
         UniqueConstraint('dag_id', 'run_id', name='dag_run_dag_id_run_id_key'),
         Index('idx_last_scheduling_decision', last_scheduling_decision),
+        # since mysql lacks filtered/partial indices, this creates a
+        # duplicate index on mysql. Not the end of the world
```

Comment:
This PR now depends on #18061 because of this comment. Will merge after merging #18061